CMakeLists.txt

```diff
@@ -679,6 +679,119 @@ if (VAST_ENABLE_MANPAGES)
   find_package(Pandoc REQUIRED)
 endif ()
 
+# -- caf -----------------------------------------------------------------------
+
+find_package(OpenSSL REQUIRED)
+
+function (VASTSystemizeTarget tgt)
+  message(STATUS "Treating ${tgt} as a system dependency")
+  get_target_property(_includes ${tgt} INTERFACE_INCLUDE_DIRECTORIES)
+  set_property(
+    TARGET ${tgt}
+    APPEND
+    PROPERTY INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${_includes}")
+endfunction ()
+
+# TODO: Require CAF to be installed.
+
+if (VAST_ENABLE_DEVELOPER_MODE
+    AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/libvast/aux/caf/CMakeLists.txt"
+    AND NOT CAF_ROOT_DIR)
+  set(VAST_ENABLE_BUNDLED_CAF_DEFAULT ON)
+else ()
+  set(VAST_ENABLE_BUNDLED_CAF_DEFAULT OFF)
+endif ()
+option(VAST_ENABLE_BUNDLED_CAF "Use the CAF submodule"
+       "${VAST_ENABLE_BUNDLED_CAF_DEFAULT}")
+add_feature_info("VAST_ENABLE_BUNDLED_CAF" VAST_ENABLE_BUNDLED_CAF
+                 "use the CAF submodule.")
+if (NOT VAST_ENABLE_BUNDLED_CAF)
+  # Try to find the required CAF components first...
+  find_package(
+    CAF
+    COMPONENTS core io test openssl
+    REQUIRED)
+  if (NOT ${CAF_VERSION} VERSION_GREATER_EQUAL 0.18.6)
+    message(
+      FATAL_ERROR "Failed to find CAF >= 0.18.6 version (found ${CAF_VERSION})")
+  endif ()
+  string(
+    APPEND
+    VAST_FIND_DEPENDENCY_LIST
+    "\nfind_package(CAF ${CAF_VERSION} COMPONENTS core io test openssl REQUIRED)"
+  )
+else ()
+  # Use bundled CAF.
+  if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/libvast/aux/caf/CMakeLists.txt")
+    message(
+      FATAL_ERROR
+        "CAF not found, either use -DCAF_ROOT_DIR=... or initialize the libvast/aux/caf submodule"
+    )
+  else ()
+    set(VAST_ENABLE_BUNDLED_CAF ON)
```

Comment:
If I understand correctly then this is the only line that is actually problematic and would need to be changed or, as this is used in `VASTConfig.cmake.in`.

It makes a lot more sense to keep the CAF include in `libvast/CMakeLists.txt` and then to fix the actual underlying issue instead of moving things up because of directory-scope variables.

Reply:
It is neither. `libvast_test` doesn't see `CAF::test` when `find_package` is called in a different branch of the directory tree.
I think it is only logical to move the dependency setup one level up since both libraries need it.

Reply:
Okay, that convinces me. Can you put that into a comment above where we load this? I believe the `find_package(OpenSSL)` can also move back to `libvast`.

Reply:
The `find_package(OpenSSL)` is needed here because it's missing from `CAFConfig.cmake`.

Reply:
Isn't it rather missing from VASTConfig.cmake? We added the OpenSSL dependency because of the pseudonymize operator, not because of CAF.

Reply:
CAF::openssl depends on openssl but CAF itself does not pull it in. Before the CAF bump this was handled by the find module, but I'd rather not keep on maintaining that.
VASTConfig.cmake might technically need to pull in OpenSSL for static libvast installations, but nobody is using such a configuration at the moment. Feel free to push a fix for that to this branch, but leave the `find_package(OpenSSL)` here.

Reply:
Makes sense. I might take a stab at cleaning this up a bit sometime next year. May also be a good idea to get a few fixes in upstream while CAF 0.19 is still in the rc-phase.